alerting/grafana_alertmanager.go

```diff
@@ -135,7 +112,32 @@ type MaintenanceOptions interface {
 	MaintenanceFunc() (int64, error)
 }
 
+type Template = template.Template
+type InhibitRule = config.InhibitRule
+type MuteTimeInterval = config.MuteTimeInterval
+type Route = config.Route
+type Integration = notify.Integration
+type DispatcherLimits = dispatch.Limits
+type Notifier = notify.Notifier
+
+// Configuration is an interface for accessing Alertmanager configuration.
+type Configuration interface {
```

Comment:
This is the biggest piece of this work - we've abstracted everything that could be Grafana-specific in an interface, there's probably more than will either go to Grafana or come back here but I've tried to do the minimum for done to get something done.